### PR TITLE
Issue/65/fix inference engine for real exam case

### LINKED ISSUE #65: Fix inference engine for real exam case

### PR DESCRIPTION
**Pull request by marekhering**
Closes #65

### CODE REVIEW

File: inference_engine/src/preprocessing/rotation/rotation.py [hunk 1]
```diff
@@ -34,18 +37,23 @@ def _get_contours(new_image):
...
     return squares
 
 
-def _set_numeric_values_squares(contours, threshold=0.01):
+def _set_numeric_values_squares(contours, threshold=20):
     def is_similar(first_delta, second_delta):
-        return abs(first_delta[0] - second_delta[0]) <= max(threshold, min(first_delta[0], second_delta[0]) / 100) and \
-               abs(first_delta[1] - second_delta[1]) <= max(threshold, min(first_delta[1], second_delta[1]) / 100)
+        return abs(first_delta[0] - second_delta[0]) <= max(threshold, min(first_delta[0], second_delta[0]) / 10) and \
+               abs(first_delta[1] - second_delta[1]) <= max(threshold, min(first_delta[1], second_delta[1]) / 10)
```
Comment (Debskij) [inline inference_engine/src/preprocessing/rotation/rotation.py:56]: [Non blocking] Lets maybe move those value to some config file

File: inference_engine/src/preprocessing/rotation/rotation.py [hunk 2]
```diff
@@ -34,18 +37,23 @@ def _get_contours(new_image):
     for c in contours:
         # get rotated rectangle from contour
         rot_rect = cv2.minAreaRect(c)
-        if abs(rot_rect[1][0] - rot_rect[1][1]) < min(rot_rect[1]) / 10:
+        (x, y), (w, h), alpha = rot_rect
+        if abs(w - h) < min((w, h)) / 5:
```
Comment (Debskij) [inline inference_engine/src/preprocessing/rotation/rotation.py:41]: [Non blocking] Lets maybe move those value to some config file

**Review by Debskij — APPROVED** [review]

File: inference_engine/src/controller.py [hunk 3]
```diff
@@ -95,22 +95,17 @@ def _check_pdf(exam_id, file_name, pdf_type: PDFType, force=False):
...
                 continue
             output_dir = tmp_dir
             sufix = ""
             if pdf_type == PDFType.answer_sheets:
                 output_dir = output_dir / results.student_id_boxes
-                version = versioning.get_next_version(
-                    output_dir,
-                    remote_storage.get_student_dir(exam_id, index=results.student_id_boxes),
-                )
-            elif pdf_type == PDFType.answer_keys:
```
Comment (leckijakub) [inline inference_engine/src/controller.py:113]: @marekhering this breaks handling of both students answers and answers keys. This hunk should be restored

**Comment by leckijakub** [comment]
@marekhering  tested with some scanned exams. Empty exam answers_1 is detected as student `110477` .
Moreover, in both filled and empty scans exam_key is detected as [1,1,1,1]
answers_3

**Review by leckijakub — CHANGES_REQUESTED** [review]

**Comment by leckijakub** [comment]
@marekhering to generate templates I changed color of letters in boxes from `darkgray` do `lightgray` in https://github.com/cvlab-ai/splinter/blob/main/research/answer_sheet_generator/constructs.py#L35

**Review by leckijakub — APPROVED** [review]
@marekhering Thanks for the updates, now the exams are correctly detected. Approve

**Pull request merged**